topography/include/topography.inc

```diff
@@ -703,272 +443,152 @@ end interface
            zdat = zdat*zdat
            call horiz_interp ( Interp, zdat(:,js:je), zout2 )
            zout = zout2 - zout*zout
-           where (zout > 0.0)
+           where (zout > 0.0_lkind)
              zout = sqrt ( zout )
            elsewhere
-             zout = 0.0
+             zout = 0.0_lkind
            endwhere
        endif
     endif
 
-    call horiz_interp_del ( Interp )
+    call horiz_interp_del( Interp )
 
- end subroutine interp_topog_2d
+end subroutine INTERP_TOPOG_2D_
 
 !#######################################################################
 
- subroutine find_indices ( ybeg, yend, ydat, js, je )
- real,    intent(in)  :: ybeg, yend, ydat(:)
- integer, intent(out) :: js, je
- integer :: j
+subroutine FIND_INDICES_( ybeg, yend, ydat, js, je )
+real(kind=FMS_TOP_KIND_), intent(in)  :: ybeg, yend, ydat(:)
+integer,                  intent(out) :: js, je
+integer                               :: j
+integer, parameter                    :: lkind = FMS_TOP_KIND_
 
    js = 1
    do j = 1, size(ydat(:))-1
-      if (ybeg >= ydat(j) .and. ybeg <= ydat(j+1)) then
+      if (ybeg >= ydat(j) .and. ybeg <= ydat(real(j,FMS_TOP_KIND_)+1.0_lkind)) then
```

Comment:
I don't think this needs a real, I think j+1 is the index of ydat

Reply:
resolved in b7400af4